rosetta-integration-tests/src/test/java/com/regnosys/rosetta/validation/ExpressionValidatorTest.java

```diff
@@ -22,6 +22,74 @@ public class ExpressionValidatorTest {
     private RosettaValidationTestHelper validationTestHelper;
     @Inject
     private RosettaTestModelService modelService;
+
+    @Test
+    void thenExpressionImplicitVariableThatDoesNotReferenceArgumentIsDisallowed() {
+        RosettaExpression expr = modelService.toTestModel("""
+                type Foo:
+                    isAllowable boolean (1..1)
+               """).parseExpression("""
+               Foo { isAllowable: False } then ("hello" extract item)
+               """);
+
+        validationTestHelper.assertError(expr, INLINE_FUNCTION, null,
+                "The input item is not used in the `then` expression");
+    }
+
+    @Test
+    void thenExpressionReturnsAttributeOfLeftHandSideWithNoIssues() {
+        RosettaExpression expr = modelService.toTestModel("""
+               
+               type Foo:
+                    isAllowable boolean (1..1)
+               """).parseExpression("""
+               Foo { isAllowable: False } then isAllowable
+               """);
+
+        validationTestHelper.assertNoIssues(expr);
+    }
+
+    @Test
+    void thenExpressionThatCallsFunctionShouldHaveNoIssues() {
+    	RosettaExpression expr = modelService.toTestModel("""
+            func SomeFunc:
+                inputs:
+                	isAllowable boolean (1..1)
+                output:
+                	result string (1..1)
+            
+                set result: if isAllowable then "allowed" else "not allowed"
+            """).parseExpression("""
+               False then SomeFunc
+               """);
+    	
+    	validationTestHelper.assertNoIssues(expr);
+    } 
+    
+    @Test
+    void thenExpressionWithItemShouldHaveNoIssues() {
+    	RosettaExpression expr = modelService.toTestModel("""
+                type Foo:
+                     isAllowable boolean (1..1)
+               """).parseExpression("""
+               Foo { isAllowable: False } filter isAllowable then extract "someResult"
+               """);
+    	
+    	validationTestHelper.assertNoIssues(expr);
+    }    
+    
+    @Test
+    void thenExpressionWithNoItemShouldBeDisallowed() {
+    	RosettaExpression expr = modelService.toTestModel("""
+                type Foo:
+                     isAllowable boolean (1..1)
+               """).parseExpression("""
+               Foo { isAllowable: False } filter isAllowable then "someResult"
+               """);
+    	
+    	validationTestHelper.assertError(expr, INLINE_FUNCTION, null,
+                "The input item is not used in the `then` expression");
+    }
```

Comment:
Might be worth adding tests with nested `item` cases.

E.g., `"input" then ("hello" extract item)` should raise an error since the `"input"` is never used, even though `item` is present.

Similarly, `"input" then (Foo { isAllowable: False } extract isAllowable)` should throw an error, even though `isAllowable` refers to an attribute of the implicit variable, but not the right one.